File: app/src/main/java/com/codekk/mvp/presenter/OpListPresenterImpl.java
package com.codekk.mvp.presenter;

import com.codekk.data.Constant;
import com.codekk.mvp.model.OpListModel;
import com.codekk.mvp.view.ViewManager;
import com.codekk.net.Api;
import com.codekk.net.NetFunc;
import com.codekk.ui.base.BasePresenterImpl;
import com.common.widget.StatusLayout;

import io.reactivex.network.manager.RxNetWork;

/**
 * by y on 2017/5/16
 */

public class OpListPresenterImpl extends BasePresenterImpl<ViewManager.OpListView, OpListModel>
        implements PresenterManager.OpListPresenter {

    public OpListPresenterImpl(ViewManager.OpListView view) {
        super(view);
    }

    @Override
    public void netWorkRequest(int page) {
        netWork(RxNetWork
                .observable(Api.OpService.class)
                .getOpList(page, Constant.OP_LIST_TYPE_ARRAY)
                .map(new NetFunc<>()));
    }

    @Override
    public void onNetWorkSuccess(OpListModel data) {
        if (data.getProjectArray().isEmpty()) {
            view.noMore();
        } else {
            super.onNetWorkSuccess(data);
        }
    }
}

File: app/src/main/java/com/codekk/mvp/presenter/OpListPresenterImpl.kt
package com.codekk.mvp.presenter

import com.codekk.NetFunc
import com.codekk.OpService
import com.codekk.mvp.model.OpListModel
import com.codekk.mvp.view.ViewManager
import com.codekk.ui.base.BasePresenterImpl
import io.reactivex.network.RxNetWork


/**
 * by y on 2017/5/16
 */
class OpListPresenterImpl(view: ViewManager.OpListView) : BasePresenterImpl<ViewManager.OpListView, OpListModel>(view), PresenterManager.OpListPresenter {

    override fun netWorkRequest(page: Int) {
        netWork(RxNetWork
                .observable(OpService::class.java)
                .getOpList(page)
                .map(NetFunc()))
    }

    override fun onNetWorkSuccess(data: OpListModel) {
        if (data.projectArray.isEmpty()) {
            view?.noMore()
        } else {
            super.onNetWorkSuccess(data)
        }
    }
}
